### ISSUE TITLE
Adapt rootcheck to new plugin: policy monitoring

### ISSUE THREAD

**Issue by jesuslinares** [wazuh/wazuh#2519]
5th Week
- Adapt `/rootcheck`.
- Create `/policymonitoring`.
- Get scans from policy monitoring.

Blocked by https://github.com/wazuh/wazuh/issues/1983.

6th Week
Still blocked

**Comment by crd1985** [comment]
# Status update
## Summary
Fix some implementation errors I discovered after building the demo environment.
I needed to fix the test database and I almost have it working. It remains an error when filtering in path with url encoded strings like Linux%20Ubuntu. Once this is working, I will have policy monitoring developed and ready for test (unless more changes happen).

## Tasks
* [3h] Fix implementation errors after manual testing.


## Stoppers
* Still https://github.com/wazuh/wazuh/issues/1983

## Difficulties
* Changing sql model
* Messy class hierarchy to deal with Db connections (WazuhDBConnection, WazuhDBQuery, ...)
* Messy node js API

## Tests performed
I need to rerun tests after changes.

## Pending
* Finish unit testing (subject to changes until sql model is stable)
* Solve issue filtering by url encoded strings (the regex does not match apparently)

**Comment by crd1985** [comment]
@wazuh/frontend New changes according to https://github.com/wazuh/wazuh/issues/2518#issuecomment-460966353

The format of the endpoints would be:

```
# GET /policy-monitoring
{
	"error": 0,
	"data": {
		"totalItems": 2,
		"items": [
			{
				"scan_id": 1,
                                "name": "pm_name1",
                                "description": "description1",
				"pass": 272,
				"fail": 33,
				"unknown": 0,
				"score": 89,
                                "policy_id": "aaa",
                                "hash": "asdfasdfasdfasdfasdf",
				"start_scan": "2019-01-11 19:51:45",
				"end_scan": "2019-01-11 20:51:45",
                                "references": "references1"
			},
			{
				"scan_id": 2,
                                "name": "pm_name2",
                                "description": "description1",
				"pass": 22,
				"fail": 3,
				"unknown": 5,
				"score": 56,
                                "policy_id": "bbb",
                                "hash": "asdfasdfasdfasdfasdf",
				"start_scan": "2019-01-11 18:51:45",
				"end_scan": "2019-01-12 18:51:45",
                                "references": "references2"
			}
		]
	}
}

# GET /policy-monitoring/000/checks/CIS%20Checks%20for%20Apache%20Https%20Server
{
	"error": 0,
	"data": {
		"totalItems": 1,
		"items": [
			{
				"id": 1,
				"title": "Status Module is enabled",
				"description": "a description",
                                "rationale": "rationale desc",
                                "remediation": "Fix that",
				"file": "/etc/http.conf",
				"process": null,
				"directory": null,
				"registry": null,
				"references": "https://workbench.cisecurity.org/benchmarks/307,https://workbench.cisecurity.org/benchmarks/308",
				"cis": "Apache Configuration - 2.4",
				"result": "fail",
                                "compliance": [
                                        {"key": "Compliance1",
                                         "value": "Valueofcompliance1"},
                                        {"key": "Compliance2",
                                         "value": "Valueofcompliance2"}
                                ]
			}
		]
	}
}
```

**Comment by crd1985** [comment]
# Status update
## Summary
Main functionality is working, but small changes in the sql model requires a new review.
Some minor fixes after merging to 3.9-alpha.

## Stoppers
* Still https://github.com/wazuh/wazuh/issues/1983

## Difficulties
* Changing sql model
* Messy class hierarchy to deal with Db connections (WazuhDBConnection, WazuhDBQuery, ...)
* Messy node js API

## Tests performed
I need to rerun tests after changes. Mocha tests cannot be run until core is done.

## Pending
* Mocha tests
* Final review
